component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java

```diff
@@ -386,9 +388,16 @@ public ComponentManager(final File m2, final String dependenciesResource, final
         this.container.registerListener(new Updater());
         ofNullable(jmxNamePattern).map(String::trim).filter(n -> !n.isEmpty())
                 .ifPresent(p -> this.container.registerListener(new JmxManager(p, ManagementFactory.getPlatformMBeanServer())));
-        this.extensions = StreamSupport.stream(Spliterators.spliteratorUnknownSize(
-                ServiceLoader.load(ComponentExtension.class, Thread.currentThread().getContextClassLoader()).iterator(),
-                Spliterator.IMMUTABLE), false).collect(toList());
+        toStream(loadServiceProviders(ContainerListenerExtension.class)).forEach(listener -> container.registerListener(listener));  
+        this.extensions =  toStream(loadServiceProviders(ComponentExtension.class)).collect(toList());
+    }
+    
+    private static <T> Stream<T> toStream(Iterator<T> iterator) {
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(iterator, Spliterator.IMMUTABLE), false);
+    }
+    
+    private static <T> Iterator<T> loadServiceProviders(Class<T> service) {
+        return ServiceLoader.load(service, Thread.currentThread().getContextClassLoader()).iterator();
```

Comment:
the classloader is defined in org.talend.sdk.component.runtime.manager.ComponentManager#ComponentManager, can you please ensure we use a single instance and don't rely on the TCCL in multiple places in case we refactor the code later on? (just define a ClassLoader var in the constructor and pass it to the loadServiceProviders)